### ISSUE TITLE
Overflow even with fp32

### ISSUE THREAD

**Issue by herokoder** [microsoft/DeepSpeed#2673]
I am running a training with deepspeed with fp16 disabled (and with fp32), but I get an error `overflow! rank 0 skipping step. attempted loss scale: 1.0 reducing to 1.0`. This warning occurs about 10% of the time and usually is not sequential in training steps. I checked my weights and gradients and all are reasonable floats with magnitude less than 100. Can anyone tell me what the problem might be? My ds_config is the following 
```dsconfig = {
            "train_batch_size": batch_size*args.world_size*args.gradient_accumulation_steps,
            "train_micro_batch_size_per_gpu": batch_size,
            "gradient_accumulation_steps": args.gradient_accumulation_steps,
            "zero_allow_untested_optimizer": True,
            "optimizer": 
                {
                "type": "AdamW",
                "params": 
                    {
                    "lr": 1.e-4,
                    "betas": [0.9, 0.999],
                    "eps": 1e-8,
                    "weight_decay": 0
                    },
                },
                "fp16": {
                    "enabled": False,
                    "min_loss_scale": 1,
                    "initial_scale_power": 32,
                    "loss_scale_window": 1000,
                    "opt_level": "O3"
                },
                "zero_optimization": {
                    "stage": 1,
                    "cpu_offload": True,
                    "contiguous_gradients": True,
                    "overlap_comm": True
                }
        }


If I enable fp16, I get a bunch of overflow messages and at the end of the first epoch the run gets killed with following message `Current loss scale already at minimum - cannot decrease scale anymore. Exiting run.`

**Comment by tohtana** [comment]
Hello @herokoder, thank you for reporting. 

I want to reproduce the problem on our environment. 
Can you provide more details about how you ran the training? 

I would like to know the model, the dataset, and the training settings including the batch size as well as the command line. 
The details of the environment (versions of Python, PyTorch and DeepSpeed, GPU models) would be also very helpful.

**Comment by kyleliang919** [comment]
@herokoder can you elaborate what caused this problem? I am facing similar issue on a customized dataset, will really appreciate it if you can give some hints on what I should look for.

**Comment by LuciusMos** [comment]
> @herokoder can you elaborate what caused this problem? I am facing similar issue on a customized dataset, will really appreciate it if you can give some hints on what I should look for.

@herokoder Hi! I met the same problem on my own dataset. Thank you very much if you would like to share some information!

**Comment by YueChenkkk** [comment]
I met the same problem when LoRA-finetuning long-t5 on the BookSum long narrative text summarization dataset.
Seems long input sequences might cause infinite gradients.
